firestore.rules

```diff
@@ -135,33 +129,138 @@ service cloud.firestore {
     match /authed/{portal} {
       allow read, write: if isAuthedTeacher();
 
+      function getTeacherNetworks() {
+        let platformUserId = request.auth.token.platform_user_id;
+        return get(/databases/$(database)/documents/authed/$(portal)/users/$(platformUserId)).data.networks;
+      }
+
+      function resourceInTeacherNetworks() {
+        return resource.data.network in getTeacherNetworks();
```

Comment:
If you are using this successfully then you probably already understand this restriction of rules:
https://firebase.google.com/docs/firestore/security/rules-query

In this case what it means is that the `network` field will need to be in the query constraints sent by the CLUE client when the document is requested. This is a really nuanced piece of firestore security rules, so I just wanted to clarify.

Reply:
>If you are using this successfully then you probably already understand this restriction of rules:

I am certainly aware of them and was trying to keep them in mind. I'm not sure I've fully grokked all of the implications and wouldn't be completely surprised if I got tripped up by them at some point. The unit tests pass and are trying to test all of these pieces so to the extent that the rules unit tests are a valid test, then I believe they're being used successfully.